### ISSUE TITLE
Bug on 0.9.42 / plus: How to use the lib with Ionic 3 / Angular 5

### ISSUE THREAD

**Issue by Charles1972** [stephanrauh/ngx-extended-pdf-viewer#86]
Hi, after npm install to new version 0.9.42, the component stop to run with this error:
-Uncaught TypeError: Object(...) is not a function
    at ngx-extended-pdf-viewer.js:880
    at Module../node_modules/ngx-extended-pdf-viewer/fesm5/ngx-extended-pdf-viewer.js (ngx-extended-pdf-viewer.js:888)
    at __webpack_require__ (bootstrap:83)
    at Module../src/app/components/pdf-container/pdf-container.component.ts (main.js:1189)
    at __webpack_require__ (bootstrap:83)
    at Module../src/app/app-routing.module.ts (main.js:79)
    at __webpack_require__ (bootstrap:83)
    at Module../src/app/app.module.ts (app.component.ts:13)
    at __webpack_require__ (bootstrap:83)
    at Module../src/main.ts (main.ts:1)

And this warning during compiling:
-WARNING in ./node_modules/ngx-extended-pdf-viewer/fesm5/ngx-extended-pdf-viewer.js 880:50-68
"export 'ɵɵdefineInjectable' was not found in '@angular/core'

Have you any idea how to solve it.

Thanks, in advance!

**Comment by skyleguy** [comment]
I am also having this issue

**Comment by stephanrauh** [comment]
Weird... it looks like there's something gone wrong in the Angular package.

![image](https://user-images.githubusercontent.com/3045767/58986273-58de5a00-87dd-11e9-85b1-44c4aeaca207.png)

Other teams see this error, too: https://github.com/valor-software/ng2-charts/issues/1115

**Comment by stephanrauh** [comment]
Also see https://github.com/angular/angular/issues/30413. Maybe it's necessary to compile the library with Angular 7.

**Comment by stephanrauh** [comment]
I've published a new version 0.9.43 that's been compiled with Angular 7. Please run a test and tell me if it fixes the problem.

Thanks in advance,
Stephan

**Comment by Charles1972** [comment]
Perfect!

Thank you for the good job!!! ;)

**Comment by tanzl88** [comment]
I still see this error on ionic 3 angular 5 running version 0.9.25.

Uncaught TypeError: Object(...) is not a function
    at ngx-extended-pdf-viewer.js:1062
    at Object.<anonymous> (ngx-extended-pdf-viewer.js:1064)

**Comment by stephanrauh** [comment]
@tanzl88 Do you see the `ɵɵdefineInjectable` bit? Otherwise, it might be another bug.

**Comment by tanzl88** [comment]
No, this is the line of code where it happens. I think is the defineInjectable bit. 

`NgxExtendedPdfViewerService.ngInjectableDef = defineInjectable({ factory: function NgxExtendedPdfViewerService_Factory() { return new NgxExtendedPdfViewerService(); }, token: NgxExtendedPdfViewerService, providedIn: "root" });`

**Comment by tanzl88** [comment]
I found defineInjectable is only introduced in angular 6. Even though it has been deprecated in angular 8.

**Comment by stephanrauh** [comment]
Reopened on request by @tanzl88.

**Comment by stephanrauh** [comment]
@tanzl88 I've created a repositoy using both Ionic 5.1 and ngx-extended-pdf-viewer 0.9.49. I can't reproduce your bug.

Please clone https://github.com/stephanrauh/ionic-5-pdf-viewer and run the example. Does it show the PDF file in the second tab? Are there any error messages? If so, there's a difference between my machine and your PC, so we can continue the investigation. If not, there's probably a difference between your project and the way I've set up the project.

Either way, I'm looking forward to your feedback!

**Comment by tanzl88** [comment]
I meant ionic 3 using angular 5. Here is the reproducer. https://github.com/tanzl88/ionic-3-extended-pdf-viewer.

The error came from defineInjectable which is only introduced in angular 6. I understand the plugin only support angular 6 and above. Is there any chance to provide angular 5 support?

**Comment by stephanrauh** [comment]
Thanks for the reproducer!

I've spent some time investigating Angular 5 and Ionic 3. As you've found out, Angular 6 introduced several breaking changes concerning libraries. On of the - ngOnChanges used to be called before applying the changes, while it now is called after applying the changes - convinced my to give up the Angular 5 dream. Now ngx-extended-pdf-viewer officially required Angular 6+.

But that's no reason to despair. ngx-extended-pdf-viewer is just a handful of files. You can easily copy them to your own project. So that's what I did. I forked your project and integrated ngx-extended-pdf-viewer: https://github.com/stephanrauh/ionic-3-extended-pdf-viewer

The good news: the project compiler. The bad news: it shows a runtime error I don't understand. I believe it's just a matter of updating the polyfills. But I didn't find out how to do that with Ionic 3.

@tanzl88 I'd be happy if you were to pick up my work and finish it. I believe we're almost there. Just 5% missing. If you manage to do so, I'd like to publish your solution on https://www.beyondjava.net and https://pdfviewer.net, preferably mentioning your real-life name, so your career gets a boost!

**Comment by tanzl88** [comment]
I got it to work on ionic 3. I have updated the repo too. https://github.com/tanzl88/ionic-3-extended-pdf-viewer. To be honest, I don't understand what causes the polyfill problem as well. But I found some thread mentioned to load external js files in index.html file. So I guess it might be an ionic thing. Anyway, this is what I did.
1. Load pdfjs in index.html
2. Copy ngx-extended-pdf-viewer into component
3. Change ngx-extended-pdf-viewer css file into ionic format (remove styleUrls)

Btw, did you do anything to the ngx-extended-pdf-viewer files? For the upcoming updates can I just copy the files into my project and use it as a component?
